src/docs/background/privileged-transitions/index.md

```diff
@@ -0,0 +1,87 @@
+published: true
+---
+
+_**NOTE! Privileged transitions enable highly flexible pricing schemes,
+deprecating our custom pricing feature that was in the past used for
+price customizations. If you are looking for the old custom pricing
+article, it can be found
+[here](https://5ee94c280d38f10008a3bfa1--sharetribe-flex-docs-site.netlify.app/docs/background/custom-pricing/).**_
+
+## What are privileged transitions?
+
+In Flex, a process transition is an edge between two states in the
+transaction process graph. Invoking transitions is guarded in the
+process definition by tying them to a specific state when they can be
+transitioned and by defining which party of a transaction can perform
+the transition. This way transition requests have built-in validation of
+who can invoke them and in what state of the transaction flow.
+
+However, there are moments when more control is required on who can
+initiate a transition and especially with what kind of parameters. Take
+discounts in pricing by leveraging discount coupons managed by a 3rd
+party service for example. A coupon code can be validated in the client
+side by invoking the coupon service but this will not limit who will be
+technically able to invoke a pricing related transition with discounted
+price parameters.
+
+This is where privileged transitions come into play. They are
+transaction process transitions that can be invoked from a secure
+context. In other words, this means that you can build your own server
+side validation that sits between your marketplace UI and the Flex
+Marketplace API. In the discount coupon example, this means that the
+discount coupon that a user has can be passed as a parameter in the
+transition request. Server side transition request validation can invoke
+a 3rd party service to verify a discount code, update pricing parameters
+accordingly, and pass those to the transition that has pricing actions
+tied to it.
+
+## How do privileged actions work?
+
+Standard authenticated Marketplace API requests require a valid access
+token obtained from the Authentication API. Privileged transitions
+differ here by requiring a special kind of _trusted_ token to
+authenticate properly. A trusted token can be obtained by exchanging a
+valid access token to a trusted one in the Authentication API by
+providing a client secret. The client secret is not to be exposed
+publicly, securing that privileged transitions can only be invoked from
+a trusted source, i.e. your own backend implementation.
+
+![Authentication flow with a trusted access token](auth-flow.png)
+
+The client secret is tied to a Marketplace API
+[application](/background/applications/). When exchanging an access
+token to a trusted one, the client secret needs to be from the same
+application as the client ID that was used to obtain the access token.
+
+Remember to never expose the client secret publicly. This enables full
+control over requests that invoke privileged transitions.
```

Comment:
I'm wondering if we do (or should) write client secret, client id, etc. with capital letters. Client Secret, Client ID.

Reply:
These haven't been written like that in anywhere else either, so it's a process of search and replace across all our UIs (meaning docs, console and so on...) if we wish to change this.

Reply:
Got it. No point changing then.